### PR TITLE
Add luacheck, repo cleanup

### PR DESCRIPTION
**Pull request by Niklp09**
- Add luacheck (and fix warnings)
- Remove/or cleanup old files

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -139,25 +138,10 @@ local update_item = function(pos, node)
...
-			pos.y = pos.y + 5 / 16
-			e:set_pos(pos)
-		else
-			e:set_rotation({x = -1.5708, y = yaw, z = 0}) -- x is pitch, 1.5708 is 90 degrees.
-		end
+		e:set_rotation({x = -1.5708, y = yaw, z = 0}) -- x is pitch, 1.5708 is 90 degrees.
 	end
 end
 
-local metal_sounds
```
Comment (Panquesito7) [inline init.lua:153]: Shouldn't this be kept?

**Review by Panquesito7 — COMMENTED** [review]
Would be nice if the minimum MT version on `mod.conf` is added as well.

File: init.lua (hunk 1 above)
Comment (Niklp09) [reply init.lua:153]: No, the metal sounds were added in 2016 (https://github.com/minetest/minetest_game/commit/fa9a34554b96d2223cce2a9a53991ebe3c5bedc0)

**Comment by Niklp09** [comment]
>Would be nice if the minimum MT version on mod.conf is added as well.

I see no reason for this, the mod supports mt 5.0 and higher

**Comment by Panquesito7** [comment]
> > Would be nice if the minimum MT version on mod.conf is added as well.
> 
> I see no reason for this, the mod supports mt 5.0 and higher

Exactly, that should be mentioned, IMO.

**Comment by Niklp09** [comment]
`min_minetest_version` is no official mod.conf parameter. (https://github.com/minetest/minetest/blob/master/doc/lua_api.md#modconf)

**Review by SmallJoker — APPROVED** [review]
LGTM. Will merge in a few days unless there are objections.

**Pull request merged**